todoApp/src/components/AddTodoForm.tsx

```diff
@@ -58,18 +58,23 @@ const AddTodoForm = ({
   //For cancel button
   const cancelHandler = () => {
     onCancelClicked(false);
-    console.log("cancelHandler");
   };
 
-  //useEffect for keyboard event
-  useEffect(() => {
-    document.addEventListener("keydown", (e: KeyboardEvent) => {
+  //For keyboard events
+  const handleKeyboardEvent = useCallback(
+    (e: KeyboardEvent) => {
       if (e.key === "Escape") {
         onCancelClicked(false);
-        console.log("ESCHandler");
       }
-    });
-  }, [onCancelClicked]);
+    },
+    [onCancelClicked]
+  );
+  useEffect(() => {
+    document.addEventListener("keydown", handleKeyboardEvent);
+    return () => {
+      document.removeEventListener("keydown", handleKeyboardEvent);
+    };
+  });
```

Comment:
Add dependencies. 
Move handler function inside the side effect. handler function depends on `onCancelClicked`, but `onCancelClicked` is again a function which will be recreated on every time it's component is rerendered. useCallback is of no use if its dependency is non primitive and is not memoized.